### PR TITLE
Promote apiserver.config.k8s.io/v1, kind=ResourceQuotaConfiguration

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
* This promotes `resourcequota.admission.k8s.io/v1beta1, kind=Configuration` to `apiserver.config.k8s.io/v1, kind=ResourceQuotaConfiguration`
  * The `apiserver.config.k8s.io` group for generic API server configuration was determined after the configuration type was introduced
  * The `Configuration` suffix is used consistently for top-level configuration types
* Docs updated in https://github.com/kubernetes/website/pull/17547

**Does this PR introduce a user-facing change?**:
```release-note
kube-apiserver: The `ResourceQuota` admission plugin configuration referenced from `--admission-control-config-file` admission config has been promoted to `apiVersion: apiserver.config.k8s.io/v1`, `kind: ResourceQuotaConfiguration` with no schema changes.
```

/cc @derekwaynecarr @deads2k @ahg-g @ravisantoshgudimetla

### CODE REVIEW

**Comment by liggitt** [comment]
/sig api-machinery
/priority important-soon
/milestone v1.17

related to https://github.com/kubernetes/kubernetes/pull/76310

**Review by ravisantoshgudimetla — APPROVED** [review]
Thank you @liggitt for working on this :)

/lgtm

File: plugin/pkg/admission/resourcequota/config_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+				LimitedResources: []resourcequotaapi.LimitedResource{
+					{APIGroup: "", Resource: "persistentvolumeclaims", MatchContains: []string{".storageclass.storage.k8s.io/requests.storage"}},
+					{APIGroup: "", Resource: "pods", MatchScopes: []corev1.ScopedResourceSelectorRequirement{
+						{ScopeName: "PriorityClass", Operator: "In", Values: []string{"cluster-services"}},
+					},
+					},
+				}},
+		},
+		{
+			name: "valid v1 config",
```
Comment (deads2k) [inline plugin/pkg/admission/resourcequota/config_test.go]: thanks

**Comment by deads2k** [comment]
Since this is logically generic, moving the group makes sense.  It does raise some questions about introducing a v2, but sparse versions *should* work.

/lgtm
/hold cancel

**Comment by liggitt** [comment]
the pull-kubernetes-node-e2e-containerd failure is known, not a flake, non-blocking, and being worked on - https://github.com/kubernetes/kubernetes/issues/85040

**Pull request merged**